### PR TITLE
Security Updates

### PR DESCRIPTION
**Pull request by manfontan**
Signed-off-by: Manuel Fontan <manuel.fontan@cognite.com>

# Description

Upgrade dependencies to fix security vulnerabilities CVE-2022-27664, CVE-2022-27664, CVE-2022-32149 reported by:
```
$ docker run aquasec/trivy image --ignore-unfixed --severity HIGH,CRITICAL --vuln-type library foundationdb/fdb-kubernetes-operator:v1.11.0

Total: 3 (HIGH: 2, CRITICAL: 1)

┌────────────────────────────────┬────────────────┬──────────┬────────────────────────────────────┬───────────────────────────────────┬──────────────────────────────────────────────────────────────┐
│            Library             │ Vulnerability  │ Severity │         Installed Version          │           Fixed Version           │                            Title                             │
├────────────────────────────────┼────────────────┼──────────┼────────────────────────────────────┼───────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│ github.com/emicklei/go-restful │ CVE-2022-1996  │ CRITICAL │ v2.9.5+incompatible                │ 2.16.0                            │ go-restful: Authorization Bypass Through User-Controlled Key │
│                                │                │          │                                    │                                   │ https://avd.aquasec.com/nvd/cve-2022-1996                    │
├────────────────────────────────┼────────────────┼──────────┼────────────────────────────────────┼───────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│ golang.org/x/net               │ CVE-2022-27664 │ HIGH     │ v0.0.0-20220127200216-cd36cc0744dd │ 0.0.0-20220906165146-f3363e06e74c │ golang: net/http: handle server errors after sending GOAWAY  │
│                                │                │          │                                    │                                   │ https://avd.aquasec.com/nvd/cve-2022-27664                   │
├────────────────────────────────┼────────────────┤          ├────────────────────────────────────┼───────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│ golang.org/x/text              │ CVE-2022-32149 │          │ v0.3.7                             │ 0.3.8                             │ golang: golang.org/x/text/language: ParseAcceptLanguage      │
│                                │                │          │                                    │                                   │ takes a long time to parse complex tags                      │
│                                │                │          │                                    │                                   │ https://avd.aquasec.com/nvd/cve-2022-32149                   │
└────────────────────────────────┴────────────────┴──────────┴────────────────────────────────────┴───────────────────────────────────┴──────────────────────────────────────────────────────────────┘
```

## Type of change

Upgrade and cleanup dependencies.

## Discussion

*Are there any design details that you would like to discuss further?* No

## Testing

*Please describe the tests that you ran to verify your changes. Unit tests?
Manual testing?*

*Do we need to perform additional testing once this is merged, or perform in a larger testing environment?*

## Documentation

*Did you update relevant documentation within this repository?* No

*If this change is adding new functionality, do we need to describe it in our user manual?* No

*If this change is adding or removing subreconcilers, have we updated the core technical design doc to reflect that?* No

*If this change is adding new safety checks or new potential failure modes, have we documented and how to debug potential issues?* N/A

## Follow-up

*Are there any follow-up issues that we should pursue in the future?* No

*Does this introduce new defaults that we should re-evaluate in the future?* Yes it adds replace command to go.mod to cleanup old dependencies from go.sum. The version should be upgraded when needed.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,6 +2,12 @@ module github.com/FoundationDB/fdb-kubernetes-operator
 
 go 1.19
 
+replace github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.12.1
```
Comment (johscheuer) [inline go.mod:5]: Have you checked which dependencies require the old versions? Maybe it's also valid to just update those dependencies instead adding the replace statements. Have you used any open-source tool to get the notifications of those CVEs?

**Review by johscheuer — COMMENTED** [review]
I'm not sure if the code base is affected by all the referenced CVEs multiple of those referenced CVEs are for older go versions.

File: go.mod (hunk 1 above)
Comment (larshagencognite) [reply go.mod:5]: This was based on scanning with trivy (https://github.com/aquasecurity/trivy), but we made a mistake and based the scan results on v1.8.1, that is why these CVEs refer to older go versions. We will update the PR based on v1.11.0 scan results.

Comment (johscheuer) [reply go.mod:5]: Thanks for doing that! Just wanted to know which tool was used to see if we can integrate that in our pipelines :)

**Comment by larshagencognite** [comment]
AFAICT there are two test failures, both failing with `timed out waiting for condition`. Is this likely to be related to the changes?

**Review by johscheuer — APPROVED** [review]
The test failures are related to our test infrastructure, which is sometimes unreliable when too many tests are running. We are working on that but for now we can ignore the test failures.

**Pull request merged**